### ISSUE TITLE
bug to PyTorch. 

### ISSUE THREAD

**Issue by nramelia2** [ultralytics/ultralytics#13323]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, I'm doing training, the training has run a few epochs but encountered an error like this. Several times I did the training, and the same error results were obtained. Is there any suggestion to fix it? Thank you

![Error bug](https://github.com/ultralytics/ultralytics/assets/100848020/99fc1485-3aa4-4994-8913-7efa95cedbf6)

**Comment by glenn-jocher** [comment]
Hello! Thank you for reaching out and providing detailed information about the issue you're encountering. 🛠️

From the error screenshot and your description, it seems like a specific problem that recurs during your training process. To better assist you, could you please provide the following additional details:

1. The exact error message or code snippet from the console (if not fully visible in the screenshot).
2. The version of PyTorch you are using, as well as any other relevant environment details (like CUDA version).
3. A brief description of your dataset and the model configuration file you're using.

With this information, we can offer more targeted advice or identify if there's a potential bug in the implementation. Meanwhile, ensure that your environment is set up according to the [installation guidelines](https://docs.ultralytics.com/install) and consider testing with a smaller subset of your data or a different model configuration to isolate the issue.

Looking forward to your response to help you resolve this issue swiftly!

**Comment by nramelia2** [comment]
![error](https://github.com/user-attachments/assets/3ea5dcc6-b3ac-48d8-93f8-440cdb7973bf)

Previously, I added another layer, as shown in the image above. When I did the training before, I tried it, and it succeeded. However, a few months later, I did it again because I wanted to add another layer, but it didn't work. Then, I tried to test it by adding a layer as I had done before. Then, an error occurred like that.

**Comment by pderrenger** [comment]
@nramelia2 thanks for sharing the details. It sounds like the issue might be related to the changes in the codebase or dependencies over time. Could you please ensure you're using the latest versions of the Ultralytics package and PyTorch? Sometimes updates can resolve such issues. If the problem persists, try running the training with the original configuration to see if the error is reproducible. This will help us pinpoint whether the issue is with the new layer or something else. Let us know how it goes!

**Comment by nramelia2** [comment]
I have tried to do training with the base, the training is safe until it is
completed. However, after adding a new layer, the training always stops at
epoch 21.

**Comment by pderrenger** [comment]
Thanks for the update. Since the base model trains fine but adding a new layer causes it to stop at epoch 21, it might be an issue with the new layer or how it's integrated. Please ensure you're using the latest versions of Ultralytics and PyTorch. If the problem persists, try simplifying the new layer or adding debug statements to pinpoint where it fails. This can help identify if it's a compatibility issue or something specific to the new layer. Let us know how it goes!

**Comment by nramelia2** [comment]
Okay, I've found the reason. It turns out that my memory is not enough if I
use batch size = 16. Thank you for your help.

Can I ask about NMS? What if I want to use DIoU loss function and CIoU NMS.
Where can I change the NMS?

Thank you

**Comment by pderrenger** [comment]
Glad you found the issue with the batch size! For using DIoU loss and CIoU NMS, you can modify the loss function and NMS settings in the model configuration files. Check the Ultralytics documentation for detailed guidance on where to make these changes. If you need further assistance, feel free to ask!

**Comment by nramelia2** [comment]
Okay, 

I successfully changed the loss function to DIoU. How can I use the DIoU loss function and CIoU for NMS? Where can I modify the NMS?

Thank you

**Comment by pderrenger** [comment]
Great to hear you successfully changed the loss function to DIoU. To use CIoU for NMS, you'll need to modify the NMS implementation in the Ultralytics codebase. Look for the NMS function in the `models/yolo.py` file and adjust it to use CIoU. Ensure you're using the latest version of the package to avoid compatibility issues. If you need further assistance, feel free to ask!

**Comment by nramelia2** [comment]
Can I ask you anything else?How do we do it so that when we train several times the results will be the same? I tried training with the same architecture, but the accuracy obtained was different.ThanksSent from my iPhoneOn 23 Jul 2024, at 21.20, Paula Derrenger ***@***.***> wrote:﻿
Great to hear you successfully changed the loss function to DIoU. To use CIoU for NMS, you'll need to modify the NMS implementation in the Ultralytics codebase. Look for the NMS function in the models/yolo.py file and adjust it to use CIoU. Ensure you're using the latest version of the package to avoid compatibility issues. If you need further assistance, feel free to ask!

**Comment by glenn-jocher** [comment]
@nramelia2 to ensure consistent results across multiple training runs, set a random seed for reproducibility. You can do this by adding `seed` to your training configuration. For example:

```python
model.train(data='path/to/your/data.yaml', seed=42)
```

This will help achieve more consistent accuracy. If you have further questions, feel free to ask!

**Comment by nramelia2** [comment]
![image](https://github.com/user-attachments/assets/91d4f1b4-8482-4464-a817-71eb32dbf4ea)
Hello, is it okay to ask questions on yolov10? 
Implemented the same module in yolov8, but why does it appear like this in yolov10?

**Comment by glenn-jocher** [comment]
@nramelia2 yes, you can ask questions about YOLOv10. Please provide more details about the issue you're encountering, including any error messages or code snippets, so we can assist you better.

**Comment by nramelia2** [comment]
![image](https://github.com/user-attachments/assets/394f22eb-8e21-4c9b-b920-65dabb810924)
I added C2f_DCN here and got error like that

**Comment by glenn-jocher** [comment]
It sounds like there might be a version conflict. Please ensure you're using the latest versions of both YOLOv8 and YOLOv10. If the issue persists, try isolating the problem by testing with a minimal setup and verifying compatibility.

**Comment by nramelia2** [comment]
Can I ask, I did training in yolov11 with 5 different loss functions. But, why the results are all the same? Previously I used yolov8 with different results. Thank you

**Comment by glenn-jocher** [comment]
It sounds like there might be an issue with how the loss functions are implemented or configured in YOLOv11. Please ensure each loss function is correctly set up and verify any shared configurations that might affect the outcomes.

**Comment by nramelia2** [comment]
In YOLOv8 I changed metrics.py, tal.py, and loss.py. I did the same thing in yolov10 and yolov11. However, the result was the same. Is there any difference for loss function in yolov10 and yolov11?

**Comment by glenn-jocher** [comment]
@nramelia2 yOLOv10 and YOLOv11 might have differences in how loss functions are integrated. Please ensure your modifications align with the specific architecture and implementation details of each version.
